platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/LatLng.java

```diff
@@ -318,24 +323,9 @@ public void writeToParcel(@NonNull Parcel out, int flags) {
    * @return distance in meters
    */
   public double distanceTo(@NonNull LatLng other) {
-    if (latitude == other.latitude && longitude == other.longitude) {
-      // return 0.0 to avoid a NaN
-      return 0.0;
-    }
-
-    final double a1 = Math.toRadians(this.latitude);
-    final double a2 = Math.toRadians(this.longitude);
-    final double b1 = Math.toRadians(other.getLatitude());
-    final double b2 = Math.toRadians(other.getLongitude());
-
-    final double cosa1 = Math.cos(a1);
-    final double cosb1 = Math.cos(b1);
-
-    final double t1 = cosa1 * Math.cos(a2) * cosb1 * Math.cos(b2);
-    final double t2 = cosa1 * Math.sin(a2) * cosb1 * Math.sin(b2);
-    final double t3 = Math.sin(a1) * Math.sin(b1);
-    final double tt = Math.acos(t1 + t2 + t3);
-
-    return GeometryConstants.RADIUS_EARTH_METERS * tt;
+    return TurfMeasurement.distance(
+      Point.fromLngLat(longitude, latitude),
+      Point.fromLngLat(other.getLongitude(), other.getLatitude())
+    ) * 1000; // convert km to meter
```

Comment:
`TurfMeasurment` exposes `TurfMeasurment#distance(Point, Point, String)` which may be a cleaner way to achieve the same thing :)